File: languages/php/Bloock/Entity/IdentityV2/CredentialReceipt.php
<?php

namespace Bloock\Entity\IdentityV2;

class CredentialReceipt
{
    private Credential $credential;
    private string $credentialId;
    private string $credentialType;

    /**
     * @param Credential $credential
     * @param string $credentialId
     * @param string $credentialType
     */
    public function __construct(Credential $credential, string $credentialId, string $credentialType)
    {
        $this->credential = $credential;
        $this->credentialId = $credentialId;
        $this->credentialType = $credentialType;
    }

    public static function fromProto(\Bloock\CredentialReceiptV2 $res): CredentialReceipt
    {
        return new CredentialReceipt(Credential::fromProto($res->getCredential()), $res->getCredentialId(), $res->getCredentialType());
    }

    /**
     * @return Credential
     */
    public function getCredential(): Credential
    {
        return $this->credential;
    }

    /**
     * @return string
     */
    public function getCredentialId(): string
    {
        return $this->credentialId;
    }

    /**
     * @return string
     */
    public function getCredentialType(): string
    {
        return $this->credentialType;
    }

    public function toProto(): \Bloock\CredentialReceiptV2
    {
        $p = new \Bloock\CredentialReceiptV2();
        $p->setCredential($this->credential->toProto());
        $p->setCredentialId($this->credentialId);
        $p->setCredentialType($this->credentialType);
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/CredentialReceipt.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents a receipt for a credential, including the credential itself, its ID, and type.
 */
public class CredentialReceipt {
  private final Credential credential;
  private final String credentialId;
  private final String credentialType;

  /**
   * Constructs an CredentialReceipt object with the specified parameters.
   * @param credential
   * @param credentialId
   * @param credentialType
   */
  public CredentialReceipt(Credential credential, String credentialId, String credentialType) {
    this.credential = credential;
    this.credentialId = credentialId;
    this.credentialType = credentialType;
  }

  public static CredentialReceipt fromProto(IdentityEntitiesV2.CredentialReceiptV2 res) {
    return new CredentialReceipt(
        Credential.fromProto(res.getCredential()), res.getCredentialId(), res.getCredentialType());
  }

  public IdentityEntitiesV2.CredentialReceiptV2 toProto() {
    return IdentityEntitiesV2.CredentialReceiptV2.newBuilder()
        .setCredential(this.credential.toProto())
        .setCredentialId(this.credentialId)
        .setCredentialType(this.credentialType)
        .build();
  }

  /**
   * Gets the credential object.
   * @return
   */
  public Credential getCredential() {
    return credential;
  }

  /**
   * Gets the ID associated with the credential.
   * @return
   */
  public String getCredentialId() {
    return credentialId;
  }

  /**
   * Gets the type of the credential.
   * @return
   */
  public String getCredentialType() {
    return credentialType;
  }
}
